### PR TITLE
Upkeep - improve messages + CI

### PR DESCRIPTION
**Pull request by olivroy**
~Tests won't pass because of #298~

Fixes #295 

Otherwise, use cli for messaging

* use testthat 3rd edition (use snapshot test for message validation + other lints)

* Update gh actions with `usethis::use_github_action()`

* Update pkgdown to use bootstrap 5 (adds a search bar)

* FIx a tidyselect deprecation warning
* Make format_*() functions redundant (no longer used in the package)

Happy to answer any questions you may have!

cli has the advantage of detecting automatically if it should do ansi output and fallbacks otherwise. This means that if you are using RStudio (or a terminal that supports links), test output is now clickable (as you can see, I was able to get rid of setup.R and teardown.R

R CMD CHECK passes cleanly locally ~except for the test failure~

### CODE REVIEW

File: tests/testthat/test-03-summarize-rodents.R [hunk 1]
```diff
@@ -243,8 +243,7 @@ test_that("rodent_species provides proper vectors or data frame", {
 
 
 
-test_that(desc = "na_conformer makes NA into `NA` in vectors and data frames",
-          code = {
+test_that(desc = "na_conformer makes NA into `NA` in vectors and data frames", {
```
Comment (olivroy) [inline tests/testthat/test-03-summarize-rodents.R:246]:
```suggestion
test_that("na_conformer makes NA into `NA` in vectors and data frames", {
```

**Comment by gmyenni** [comment]
Thanks @olivroy ! Can you rerun?

**Comment by olivroy** [comment]
You can accept workflow run to see if everything works! I  will run locally

**Comment by olivroy** [comment]
@gmyenni Confirm that everything passes cleanly on my end. Can't reproduce the failures encountered on CI.

**Comment by olivroy** [comment]
@gmyenni I just noticed that you weren't testing on macOS before. I will revert that change.

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -22,16 +20,18 @@ jobs:
       fail-fast: false
       matrix:
         config:
+          - {os: macos-latest,   r: 'release'}
```
Comment (olivroy) [inline .github/workflows/R-CMD-check.yaml:23]:
```suggestion
```

**Review by henrykironde — APPROVED** [review]

**Comment by olivroy** [comment]
For the R-devel checks to pass, I suspect we need 049f67c9c963c28228dd8b20508860788444c2ed and 761423b3df17fc70c510bf7e3cfbf650a77b0f29 from #302 to be added

**Comment by henrykironde** [comment]
I will merge these changes and we solve the na.strings = c("", "NA"), in a new PR

**Comment by olivroy** [comment]
@henrykironde there may be one other action required from you regarding the test-coverage action. codecov now requires a token to be set. You can set a repo or organization token by following the instructions here

https://github.com/r-lib/actions/issues/834

**Pull request merged**